include/galsim/ProbabilityTree.h

```diff
@@ -170,19 +175,22 @@ namespace galsim {
                     // stops quickly with the large flux Elements on the left.
                     double half_tot = absFlux/2.;
                     double leftSum=0.;
-                    for (; leftSum < half_tot; ++mid) leftSum += std::abs((*mid)->getFlux());
+                    for (; leftSum < half_tot && mid < end; ++mid) {
```

Comment:
This specifically is where the bug was.  Added `&& mid < end` here, since the `leftSum` with rounding can remain less than `half_tot`, even though 2*half_tot is supposed to be the sum of everything (in exact arithmetic).